utils/global-state-update-gen/src/auction_utils.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use casper_engine_test_support::internal::LmdbWasmTestBuilder;
+use casper_execution_engine::shared::stored_value::StoredValue;
+use casper_types::{
+    system::auction::{Bid, SeigniorageRecipient, SeigniorageRecipientsSnapshot},
+    AsymmetricType, EraId, Key, PublicKey, U512,
+};
+
+use crate::utils::ValidatorsDiff;
+
+/// Generates a new `SeigniorageRecipientsSnapshot` based on:
+/// - The list of validators, in format (validator_public_key,stake), both expressed as strings.
+/// - The starting era ID (the era ID at which the snapshot should start).
+/// - Count - the number of eras to be included in the snapshot.
```

Comment:
So there will be `count` many eras with the same list of validators that starts at `starting_era_id`?

Reply:
Yes. It's kind of similar to what happens at genesis - the first `auction_delay` eras are being seeded with the set of genesis validators. Here, we just re-seed all the eras that already exist in the auction contract state.